pkg/daemons/control/server.go

```diff
@@ -837,12 +836,24 @@ func waitForAPIServer(ctx context.Context, runtime *config.ControlRuntime) error
 		return err
 	}
 
-	select {
-	case <-ctx.Done():
-		return ctx.Err()
-	case err := <-promise(func() error { return app2.WaitForAPIServer(k8sClient, 5*time.Minute) }):
-		return err
-	}
+	done := make(chan struct{})
+	runtime.APIServerReady = done
+
+	go func() {
+		defer close(done)
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-promise(func() error { return app2.WaitForAPIServer(k8sClient, 5*time.Minute) }):
```

Comment:
It seems a shame to me to lose this error. Is it never useful? Should we at least log it?

Reply:
Will do

Reply:
Oh, now I remember why I don't handle the error.  This method will never return an error because it will only ever return an error on timeout but timeout isn't possible because the <-time.After will always complete before.  But now I realize there several stupid things about that.... Changes incoming :)